common/src/main/java/io/pravega/common/util/ByteArraySegment.java

```diff
@@ -376,6 +388,75 @@ public BufferView readSlice(int length) {
                 throw new OutOfBoundsException();
             }
         }
+
+        
+        /**
+         * Copied from DataInputStream.
+         * @throws UTFDataFormatException 
+         */
+        @Override
+        public String readUTF() throws UTFDataFormatException {
+            int utflen = readShort() & 0x0000FFFF;
```

Comment:
I think if we make the optimization in `AbstractBufferView` then we do not need this nearly identical method here.

Reply:
See my earlier comment.